File: C++/Demos/AnalogRead/AnalogRead.cpp
// AnalogRead.cpp : Defines the entry point for the console application.
//

#include "ConnectionService.h"
#include "TreehopperUsb.h"
#include <string>
#include <vector>
#include <iostream>

using namespace std;
using namespace Treehopper;

int main()
{
	TreehopperUsb& board = ConnectionService::instance().boards[0];
	board.connect();

	wcout << "Connected to " << board.toString() << endl;

	board.pins[5].mode(PinMode::AnalogInput);

	for (int i = 0; i<50; i++)
	{
		wcout << "Analog value: " << board.pins[5].analogVoltage() << endl;
		this_thread::sleep_for(chrono::milliseconds(100));
	}
	board.disconnect();
	return 0;
}

File: Java/demos/desktop/analogRead/src/main/java/io/treehopper/desktop/demos/AnalogRead.java
package io.treehopper.desktop.demos;

import io.treehopper.ConnectionService;
import io.treehopper.*;
import io.treehopper.Pin;
import io.treehopper.enums.PinMode;

public class AnalogRead {
    public static void main(String[] args) throws InterruptedException {
        while(true) {
            if(ConnectionService.getInstance().getBoards().size() == 0) continue;
            TreehopperUsb board = ConnectionService.getInstance().getBoards().get(0);
            board.connect();
            Pin adcPin = board.pins[10];
            adcPin.setMode(PinMode.AnalogInput);

            while (true) {
                double value = adcPin.getAnalogVoltage();
                System.out.println(value);
                Thread.sleep(100);
            }
        }
    }
}
